lost/src/main/java/com/mapzen/android/lost/internal/GeofencingApiImpl.java

```diff
@@ -16,32 +19,63 @@
 public class GeofencingApiImpl implements GeofencingApi {
 
   private final LocationManager locationManager;
+  private HashMap<String, PendingIntent> pendingIntentMap;
 
   GeofencingApiImpl(Context context) {
     locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+    pendingIntentMap = new HashMap<>();
   }
 
+  @RequiresPermission(anyOf = {Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION})
   @Override
   public void addGeofences(GeofencingRequest geofencingRequest, PendingIntent pendingIntent)
       throws SecurityException {
-    ParcelableGeofence geofence = (ParcelableGeofence) geofencingRequest.getGeofences().get(0);
-    locationManager.addProximityAlert(
-        geofence.getLatitude(),
-        geofence.getLongitude(),
-        geofence.getRadius(),
-        geofence.getDuration(),
-        pendingIntent);
+    List<Geofence> geofences = geofencingRequest.getGeofences();
+    addGeofences(geofences, pendingIntent);
   }
 
-  @Override public void addGeofences(List<Geofence> geofences, PendingIntent pendingIntent) {
-    throw new RuntimeException("Sorry, not yet implemented");
+  @RequiresPermission(anyOf = {Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION})
+  @Override public void addGeofences(List<Geofence> geofences, PendingIntent pendingIntent)
```

Comment:
Looking at `GeofencingApiTest`, I see we could add a test for this method which is similar to one that currently exists (`GeofencingApiTest#addGeofences_shouldAddProximityAlert`):

```java
@Test public void addGeofencesArray_shouldAddProximityAlert() {
    Geofence geofence = new Geofence.Builder()
        .setRequestId("test_id")
        .setCircularRegion(1, 2, 3)
        .setExpirationDuration(NEVER_EXPIRE)
        .build();
    Geofence anotherGeofence = new Geofence.Builder()
        .setRequestId("test_id_1")
        .setCircularRegion(4, 5, 6)
        .setExpirationDuration(NEVER_EXPIRE)
        .build();
    ArrayList<Geofence> geofences = new ArrayList<>();
    geofences.add(geofence);
    geofences.add(anotherGeofence);
    PendingIntent intent = Mockito.mock(PendingIntent.class);
    geofencingApi.addGeofences(geofences, intent);
    Mockito.verify(locationManager, times(1)).addProximityAlert(1, 2, 3, NEVER_EXPIRE, intent);
    Mockito.verify(locationManager, times(1)).addProximityAlert(4, 5, 6, NEVER_EXPIRE, intent);
  }
```

What do you think about adding this?